api/src/main/java/org/apache/iceberg/expressions/Projections.java

```diff
@@ -230,14 +243,23 @@ private StrictProjection(PartitionSpec spec, boolean caseSensitive) {
     @Override
     @SuppressWarnings("unchecked")
     public <T> Expression predicate(BoundPredicate<T> pred) {
-      PartitionField part = spec().getFieldBySourceId(pred.ref().fieldId());
-      if (part == null) {
+      Collection<PartitionField> parts = spec().getFieldsBySourceId(pred.ref().fieldId());
+      if (parts == null) {
         // the predicate has no partition column
         return alwaysFalse();
       }
 
-      UnboundPredicate<?> result = ((Transform<T, ?>) part.transform())
-          .projectStrict(part.name(), pred);
+      Expression result = Expressions.alwaysFalse();
+      for (PartitionField part : parts) {
+        // consider (ts > 2019-01-01T01:00:00) with date(ts) and hour(ts)
+        // projections: d >= 2019-01-02 and h >= 2019-01-01-02 (note the inclusive bounds).
+        // any timestamp where either projection predicate is true must match the original
+        // predicate. For example, ts = 2019-01-01T03:00:00 matches the hour projection but not
+        // the day, but does match the original predicate.
```

Comment:
Would it be difficult to translate the comment section into a unit test?

Reply:
See above.